src/app/bundles/bundle-page/bundle-page.component.ts

```diff
@@ -1,15 +1,30 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, Input } from '@angular/core';
+import { BundleDocument, State } from '../shared/bundle.interfaces';
+import { Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { StitchBundle } from './bundle-page.actions';
+import { ActivatedRoute, ParamMap } from '@angular/router';
 
 @Component({
   selector: 'app-bundle-page',
-  templateUrl: './bundle-page.component.html',
-  styleUrls: ['./bundle-page.component.scss']
+  templateUrl: './bundle-page.component.html'
 })
-export class BundlePageComponent implements OnInit {
+export class BundlePageComponent {
 
-  constructor() { }
+  @Input() documents$: Observable<BundleDocument[]>;
+  @Input() bundleId: ParamMap;
+  @Input() bundleTitle: string;
+  @Input() bundleDescription: string;
 
-  ngOnInit() {
+  constructor(private store: Store<{ bundle: State }>,
+              private route: ActivatedRoute) {
+    this.documents$ = this.store.select(state => state.bundle.documents);
```

Comment:
would this not be best to initialise on ngOnInit, and only declarations here

Reply:
it's just a declaration at this stage, it doesn't contain any logic that should be executed post init